### PR TITLE
Add expanded view on code snippets

### PR DESCRIPTION
**Pull request by esonhey**
Refactored the Code component 
 - Removed `element` property since there was no usage of it in our codebase
 - Removed `isColorSyntax` property since we can handle it with the `language` property, if the language property is passed so we colorize it
 - Removed the padding property, it's better to add the spacing where we are using it instead of having it there.
 - Added two missing properties in story book stories so we can test it there better.

Added an ExpandViewer component to handle the logic of expanded view generally
 - For the expand button didn't use the rebass Button component since we gonna use another styling tool soon, also I didn't use the usual .css format since I didn't see anywhere that we have used the CSS file.

Added Expanded View capability for Code component

### CODE REVIEW

**Comment by giardiv** [comment]
Task linked: [CU-1ptc2at Add expanded view on code snippets](https://app.clickup.com/t/1ptc2at)

File: src/components/code/index.tsx
Comment (esonhey) [inline src/components/code/index.tsx]: @giardiv 
what would be the default width of the popup window if the code line is too long?
I have set it to 700px by default, forgot to ask about it before raising the PR

Comment (esonhey) [reply src/components/code/index.tsx]: changed to have it as relative to window width

File: src/index.ts
Comment (snqb) [inline src/index.ts:179]: let's export the props too

File: src/components/expand-viewer/index.tsx
Comment (snqb) [inline src/components/expand-viewer/index.tsx]: nit: do we need this comment? It doesn't seem to serve any value, as the file name and the component name speak for themselves

File: src/components/expand-viewer/index.tsx
Comment (snqb) [inline src/components/expand-viewer/index.tsx:25]: there is `usePopup`, which is very suitable for cases like this.
Have you considered using it?

File: src/components/expand-viewer/index.tsx
Comment (snqb) [inline src/components/expand-viewer/index.tsx]: can we use `<IconButton />` here?
No children here

File: src/components/expand-viewer/expand-viewer.styles.ts
Comment (snqb) [inline src/components/expand-viewer/expand-viewer.styles.ts]: is it? We have  plenty of usages of `:hover` in the project

e.g. `src/components/row/group/row-group.styles.ts`

File: src/components/expand-viewer/expand-viewer.styles.ts
Comment (snqb) [inline src/components/expand-viewer/expand-viewer.styles.ts]: you don't need to access `theme` programmatically
since we use `emotion-theming`, you can just but `borderColor: 'primary'`. This is the case for every color usage

File: src/components/code/index.tsx
Comment (snqb) [inline src/components/code/index.tsx]: nit: avoid `!` where you can just flip the conditions
it makes code 1% more difficult to read, which we don't need

File: src/components/code/index.tsx
Comment (snqb) [inline src/components/code/index.tsx]: why spread here? Can it be `sx={buttonStyles}`?

File: src/components/code/index.tsx
Comment (snqb) [inline src/components/code/index.tsx:62]: I think it's alright this way
props should never be passed in non-shallow way

File: src/components/code/index.tsx
Comment (snqb) [inline src/components/code/index.tsx:188]: I know that this is a piece of old code, but since you're in the context: don't we need to call the `downloadCallback` in all cases? Now it's either save to file or call callback. 
Shouldn't it just save even if the callback is there?
@giardiv

File: src/components/code/index.tsx
Comment (snqb) [inline src/components/code/index.tsx]: I know this is old code too, but it breaks conventions not to use `use-` prefix for anything besides hooks
can you rename this in the PR?

File: src/components/code/index.tsx
Comment (snqb) [inline src/components/code/index.tsx]: is it a pattern you like with specifying default export and then below writing other components that are part of the main one?
it looks neat, but I'm not 100% sure

**Review by snqb — CHANGES_REQUESTED** [review]
actually, a very good job
I like the approach with the `ExpandViewer` 
but there are quite a few problems, mostly these are caused by you being new to the codebase 😺 

but I left the comments

File: src/components/code/index.tsx
Comment (esonhey) [reply src/components/code/index.tsx]: Yes, it's the kinda a practice I like. if the components and the main default exported component is not big I would keep them in the same file. 
It depends on how you guys are comfortable with it, if not I can separate them to their own file?

File: src/components/code/index.tsx
Comment (esonhey) [reply src/components/code/index.tsx:188]: I see in the code somewhere we are using just the `donwloadButton` prop without specifying `downloadCallback`.
https://github.com/logicalclocks/hopsworks-front/blob/1025acf20ec578558d8ca509694651d6bb5bc158/src/pages/project/training-dataset/forms/Features/FeatureTransformations/ViewTransformationPopup.tsx#L34-L45

File: src/components/code/index.tsx
Comment (esonhey) [reply src/components/code/index.tsx]: good one, sure will update the name

File: src/components/code/index.tsx
Comment (esonhey) [reply src/components/code/index.tsx]: good point, old stuff but gonna fix this ;)

File: src/components/code/index.tsx
Comment (esonhey) [reply src/components/code/index.tsx]: good one! will fix it.
actually, I was just refactoring, and moving out the component didn't touch the inside logic that much.

File: src/components/expand-viewer/expand-viewer.styles.ts
Comment (esonhey) [reply src/components/expand-viewer/expand-viewer.styles.ts]: yeah I see, since @giardiv told me to not go with rebass stuff I tried making my own button component, and unfortunately the :hover selector doesn't not work in plain react inline CSS . but I think we should still keep with rebass until we pick a new stack

File: src/components/expand-viewer/expand-viewer.styles.ts
Comment (esonhey) [reply src/components/expand-viewer/expand-viewer.styles.ts]: same reason to avoid using styling from emotion CSS. gonna update to use rebass instead.

File: src/components/expand-viewer/index.tsx
Comment (esonhey) [reply src/components/expand-viewer/index.tsx]: yeah forgot to remove it, it's here since first I developed this component inside the Code component then later move it out to its own module.

File: src/components/expand-viewer/index.tsx
Comment (esonhey) [reply src/components/expand-viewer/index.tsx:25]: yeah checked that but it's not suitable for this case since the second returned is the handleToggle which we don't need here.

File: src/index.ts
Comment (esonhey) [reply src/index.ts:179]: done

**Comment by giardiv** [comment]
It's pixel-perfect, nicely done

Waiting for last code approval bf merging it

**Review by giardiv — APPROVED** [review]

File: src/components/expand-viewer/expand-viewer.styles.ts
Comment (snqb) [reply src/components/expand-viewer/expand-viewer.styles.ts]: yeah, just use rebass stuff until we get rid off it. We just don't know when that happens yet, as it's a very big change to be made
until that, rebass is doing its job

**Review by snqb — APPROVED** [review]
very good!

**Pull request merged**
